src/GalleryTools/Commands/BackfillTfmMetadataCommand.cs

```
@@ -0,0 +1,81 @@
+    {
+        protected override string MetadataFileName => "tfmMetadata.txt";
+        
+        protected override MetadataSourceType SourceType => MetadataSourceType.Entities;
+        
+        protected override string QueryIncludes => $"{nameof(Package.SupportedFrameworks)}";
+
+        protected override int LimitTo => 100000;
+
+        protected override bool UpdateNeedsContext => true;
+
+        public static void Configure(CommandLineApplication config)
+        {
+            Configure<BackfillTfmMetadataCommand>(config);
+        }
+
+        protected override string ReadMetadata(IList<string> files, NuspecReader nuspecReader)
+        {
+            var supportedTFMs = string.Empty;
+            if (_packageService == null)
+            {
+                return supportedTFMs;
+            }
+
+            var supportedFrameworks = _packageService.GetSupportedFrameworks(nuspecReader, files);
+            foreach (var tfm in supportedFrameworks)
+            {
+                supportedTFMs += supportedTFMs == string.Empty
+                    ? tfm.GetShortFolderName()
+                    : $";{tfm.GetShortFolderName()}";
+            }
+
+            return supportedTFMs;
+        }
+
+        protected override bool ShouldWriteMetadata(string metadata) => true;
+
+        protected override void ConfigureClassMap(PackageMetadataClassMap map)
+        {
+            map.Map(x => x.Metadata).Index(3);
+        }
+
+        protected override void UpdatePackage(EntitiesContext context, Package package, string metadata)
+        {
+            var existingTFMs = package.SupportedFrameworks == null
+                ? Enumerable.Empty<string>()
+                : package.SupportedFrameworks.Select(f => f.FrameworkName.GetShortFolderName()).OrderBy(f => f);
+
+            var newTFMs = string.IsNullOrEmpty(metadata)
+                ? Enumerable.Empty<string>() 
+                : metadata.Split(';').OrderBy(f => f);
```

Might be worth noting that `;` does not appear in any frameworks so it is a safe delimiter ... right now (which means it's fine for backfill).

To be defensive we should throw an exception in `ReadMetadata` if a framework contains a `;`. Or we can refactor to allow metadata to be a List as well but that seems like too much work for a backfill job.

---

Oh wait is see `BackfillCommand<string>`. Could we make this `BackfillCommand<List<string>>`  or `BackfillCommand<List<NuGetFramework>>`? It would save this strange string encoding stuff.

---

This is much better. Thanks.